presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -759,6 +759,18 @@ public void finishDelete(Session session, TableHandle tableHandle, Collection<Sl
     @Override
     public Optional<ViewDefinition> getView(Session session, QualifiedObjectName viewName)
     {
+        ConnectorViewDefinition connectorViewDefinition = getConnectorViewDefinition(session, viewName);
+        if (connectorViewDefinition != null) {
+            return Optional.of(deserializeView(connectorViewDefinition.getViewData()));
+        }
+        else {
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    public ConnectorViewDefinition getConnectorViewDefinition(Session session, QualifiedObjectName viewName)
```

Comment:
Don't add a new method here, as this causes the engine to ask the connector about a view twice. Instead, we should add a new composite object (`ViewInfo`?) that holds the `ViewDefinition` and comment.